### DISCUSSION TITLE
Help understanding AssertionError: input size torch.Size([1, 3, 384, 640]) not equal to max model size (1, 3, 640, 640)

### DISCUSSION THREAD

**Discussion by MM-N** [ultralytics/yolov5#10802] [Q&A]
Hi,

Assertion in common.py line 537.
```
 s = self.bindings['images'].shape
 assert im.shape == s, f"input size {im.shape} {'>' if self.dynamic else 'not equal to'} max model size {s}"
```

I've searched as much as I could to find something similar but I have had no luck.
I'm new to this world and I don't really know what this error means exactly or how to fix this? Is it a mismatch of training image size vs what it's expecting? From the error are my images not dynamic?

I trained my model using the --img 640 flag.
I exported to TensorRT and saw the following in the output of the terminal:

```
TensorRT: input "images" with shape(1, 3, 640, 640) DataType.HALF
TensorRT: output "output0" with shape(1, 25200, 10) DataType.HALF
```
But when I go to use the model I get the above exception.

It's hit with this line in the script 
```
screenshot = Screen_Shot(x0 + x_corr, y0 + y_corr, w, h)
screenshot = cv.cvtColor(screenshot, cv.COLOR_BGR2RGB)
results = model(screenshot.copy()) //this is where the exception is thrown
```

Where screenshot is:
```
x0, y0, _, _ = window_rect
x_corr = 8
y_corr = 31
w, h = 1920, 1080

def Screen_Shot(left=0, top=0, width=1920, height=1080):
	stc = mss.mss()
	scr = stc.grab({
		'left': left,
		'top': top,
		'width': width,
		'height': height
	})

	img = np.array(scr)
	img = cv.cvtColor(img, cv.IMREAD_COLOR)

	return img
```
Is this the screenshot size being wrong? Or is there something else I've done wrong/need to look at?

Python 3.9
CUDA 11.7.1
cuDNN 8.6.0
Windows 11

I did see this when exporting if that's something to look at 
`TensorRT was linked against cuDNN 8.6.0 but loaded cuDNN 8.5.0`

Thanks

**Comment by puertea-z** [comment]
Getting the same error so far,  getting exactly the same error..
Don't need to worry about the "TensorRT was linked against cuDNN 8.6.0 but loaded cuDNN 8.5.0" warning. Seems like torch is loaded with cuDNN 8.5 by default. 
You can resolve it by replacing the existing 7 Cudnn DLL files in `C:\Users\<user>\anaconda3\envs\<env name>\Lib\site-packages\torch\lib` with the files in `<install path>\cudnn-windows-x86_64-8.6.0.163_cuda11\bin`
You can check it in python using the following (if you see 8600 then it's fixed).: 
```
    import torch
    print(torch.version)
    print(torch.version.cuda)
    print(torch.backends.cudnn.version())
```

But that didn't solve the error lol 

Here is the stuff I'm currently running with -
Python 3.9
CUDA 11.7.1
cuDNN 8.6.0
TensorRT 8.5.2.2
Windows 11

**Comment by Luis-Munu** [comment]
I'm finding the exact same issue, the YOLOv5 model itself is able to work with inputs of different sizes, however exporting it to TensorRT seems to enforce a 640x640 input size.

Python 3.10
CUDA 11.7.1
cuDNN 8.6.0
TensorRT 8.5
Windows 10

**Comment by Captain1986** [comment]
same error

**Comment by luizconrado** [comment]
Guys, I am having the same error.
Any luck anyone? How can we solve that?

**Comment by sza0415** [comment]
I had the same problem, but I solved it, my suggestion is that you can set multiple breakpoints to see. The reason why I caused this problem is that I was building the flask web page and used the code of the previous yolov5. The loadimage function used was the code in the original dataset, and the reasoning of the latest version of yolo5-7.0 was used. My solution is , replace the loadimage function, and use the loadimage function in the latest version of dataloader, hoping to help you.

**Comment by KXHH2021** [comment]
I am using YOLOv8 and I have the same problem, do you have any solution?

**Comment by shfunc** [comment]
U need to make the resolution of your input 640x640. Just make transform function:

```
import torchvision.transforms as T
def preprocess_for_classification(image, size=640):
    transform = T.Compose([
        T.Resize(size),
    ])
```

 or retrain the model for a lower one.
